### PR TITLE
The colors of the icons from the status bar should be in contrast with the status bar background color

### PR DESCRIPTION
**Pull request by iorgamgabriel**
Fixes https://bugzilla.mozilla.org/show_bug.cgi?id=1795650

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/main/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/main/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

Fixes #1795650
Fixes #1795650
Fixes #1795650
Fixes #1795650
Fixes #1795650
Fixes #1795650
Fixes #1795650
Fixes #1795650
Fixes #1795650
Fixes #1795650
Fixes #1795650

### CODE REVIEW

**Comment by Amejia481** [comment]
👋🏽 
@iorgamgabriel  could you add some details about how this PR connect with https://github.com/mozilla-mobile/android-components/issues/12894? I don't have the context around it. Thanks in advance!

**Comment by Amejia481** [comment]
Should we also add test to [WindowTest](https://github.com/mozilla-mobile/android-components/blob/main/components/support/ktx/src/test/java/mozilla/components/support/ktx/android/view/WindowTest.kt)?

**Comment by iorgamgabriel** [comment]
@Amejia481 it's not connected to https://github.com/mozilla-mobile/android-components/issues/12894

**Comment by Amejia481** [comment]
Thanks for the clarification, I think we will need to connect the issue # and update the commit description with it :)

**Comment by iorgamgabriel** [comment]
@Amejia481 I can't write unittests for this, because
 fun Window.getWindowInsetsController(): WindowInsetsControllerCompat  it is an extension function and I can't mock it.

**Comment by Amejia481** [comment]
> @Amejia481 I can't write unittests for this, because fun Window.getWindowInsetsController(): WindowInsetsControllerCompat it is an extension function and I can't mock it.

We could write tests to verify the expected behaviour, as we could read the value of `getWindowInsetsController().isAppearanceLightStatusBars` to verify if it has the correct one after calling `setStatusBarTheme`.

**Comment by Amejia481** [comment]
@mcarare could take a look as you have context around the previous change? :)

**Review by mcarare — APPROVED** [review]

File: components/support/ktx/src/main/java/mozilla/components/support/ktx/android/view/Window.kt [hunk 1]
```diff
@@ -17,7 +17,7 @@ import mozilla.components.support.utils.ColorUtils.isDark
  */
 fun Window.setStatusBarTheme(@ColorInt color: Int) {
```
Comment (mcarare) [inline components/support/ktx/src/main/java/mozilla/components/support/ktx/android/view/Window.kt:18]: The argument name could be renamed to `toolbarColor` so it should be clear that the decision to have a light stats bar or not is made based on the custom tab toolbar color.

File: components/support/ktx/src/main/java/mozilla/components/support/ktx/android/view/Window.kt [hunk 2]
```diff
@@ -27,7 +27,7 @@ fun Window.setStatusBarTheme(@ColorInt color: Int) {
  */
 fun Window.setNavigationBarTheme(@ColorInt color: Int) {
```
Comment (mcarare) [inline components/support/ktx/src/main/java/mozilla/components/support/ktx/android/view/Window.kt:28]: Same note as for `setStatusBarTheme` method argument.

File: components/support/ktx/src/main/java/mozilla/components/support/ktx/android/view/Window.kt (hunk 1 above)
Comment (iorgamgabriel) [reply components/support/ktx/src/main/java/mozilla/components/support/ktx/android/view/Window.kt]: Done

**Comment by iorgamgabriel** [comment]
getWindowInsetsController().isAppearanceLightStatusBars always returns false does't matter if I send Black or White , I think it is a problem with Robolectric . I made some gui tests for this . @Amejia481

File: components/support/ktx/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -2,4 +2,10 @@
    - License, v. 2.0. If a copy of the MPL was not distributed with this
    - file, You can obtain one at http://mozilla.org/MPL/2.0/. -->
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
-    package="mozilla.components.support.ktx" />
+    package="mozilla.components.support.ktx" >
+
+    <application>
+        <activity android:name=".MainActivity" />
```
Comment (Amejia481) [inline components/support/ktx/src/main/AndroidManifest.xml:8]: I think we should mark this as [exported false](https://developer.android.com/guide/topics/manifest/activity-element) as this is only for testing.

File: components/support/ktx/src/androidTest/java/mozilla/components/support/ktx/android/view/WindowKtTest.kt [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+import org.junit.Assert.assertTrue
+import org.junit.Rule
+import org.junit.Test
+
+class WindowKtTest {
+    @get:Rule
+    val activityRule: ActivityScenarioRule<MainActivity> = ActivityScenarioRule(MainActivity::class.java)
+
+    @Test
+    fun When_A_Light_Color_Is_Applied_To_Status_Bar_Then_Set_Is_Appearance_Light_Status_Bars_To_True() {
```
Comment (Amejia481) [inline components/support/ktx/src/androidTest/java/mozilla/components/support/ktx/android/view/WindowKtTest.kt:20]: Nit: For other UI tests we are using camel case convention see [SmokeTests](https://github.com/mozilla-mobile/android-components/blob/2d5d1ef117551ebd69c5d52582b59cb59498e9f9/samples/browser/src/androidTest/java/org/mozilla/samples/browser/SmokeTests.kt#L51) :)

**Review by Amejia481 — CHANGES_REQUESTED** [review]
For adding a new UI/Instrumentation tests we need add an entry to the [kind.yml ](https://github.com/mozilla-mobile/android-components/blob/91659e23ccea0fa1aa284ae48bce485b23d98a9b/taskcluster/ci/test/kind.yml#L59-L65) file, after it is added you could give it a try by calling `bors try` :)

File: components/support/ktx/src/main/AndroidManifest.xml (hunk 3 above)
Comment (iorgamgabriel) [reply components/support/ktx/src/main/AndroidManifest.xml]: Done

File: components/support/ktx/src/main/java/mozilla/components/support/ktx/MainActivity.kt [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.support.ktx
+
+import android.app.Activity
+
+/**
+ * Empty activity only to be used in UI tests.
+ */
+class MainActivity : Activity()
```
[the remark this answers is not in the document]
Comment (iorgamgabriel) [reply components/support/ktx/src/main/java/mozilla/components/support/ktx/MainActivity.kt]: Done

File: components/support/ktx/src/androidTest/java/mozilla/components/support/ktx/android/view/WindowKtTest.kt (hunk 4 above)
Comment (iorgamgabriel) [reply components/support/ktx/src/androidTest/java/mozilla/components/support/ktx/android/view/WindowKtTest.kt]: Done

**Comment by iorgamgabriel** [comment]
bors try

**Comment by iorgamgabriel** [comment]
bors try

**Comment by iorgamgabriel** [comment]
The tests are failed because they run on API Level 21 and 
    /**
     * Checks if the foreground of the status bar is set to light.
     * <p>
     * This method always returns false on API < 23.
     *
     * @return true if the foreground is light
     * @see #setAppearanceLightStatusBars(boolean)
     */
    public boolean isAppearanceLightStatusBars() {
        return mImpl.isAppearanceLightStatusBars();
    }

File: components/support/ktx/src/main/AndroidManifest.xml [hunk 6]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (Amejia481) [inline components/support/ktx/src/main/AndroidManifest.xml:1]: nit: Extra line.

Comment (iorgamgabriel) [reply components/support/ktx/src/main/AndroidManifest.xml]: Done

**Comment by Amejia481** [comment]
bors try

**Comment by iorgamgabriel** [comment]
bors try

**Review by Amejia481 — APPROVED** [review]
Great job 🎖️ !

**Pull request merged**